package/MDAnalysis/lib/c_gridsearch.pyx

```diff
@@ -0,0 +1,647 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+
+# Preprocessor DEFs
+DEF DIM = 3
+DEF XX = 0
+DEF YY = 1
+DEF ZZ = 2
+DEF RET_OK = 1
+DEF RET_ERROR = 0
+DEF EPSILON = 1e-5
+DEF NEIGHBORHOOD_ALLOCATION_INCREMENT = 50
+DEF GRID_ALLOCATION_INCREMENT = 50
+
+DEF BOX_MARGIN=1.0010
+DEF MAX_NTRICVEC=12
+
+from libc.stdlib cimport malloc, realloc, free, abort

```

Comment:
Are calls to `abort()` good style for cython? Will calling code be able to properly handle them? What kinds of exceptions do they raise?

(I have a few comments on abort below but I didn't flag all of them.)

Reply:
After reading more about `abort` it seems to close the program In the context of Fatslim, it is what you want, but in that context it means closing the python interpretor. Returning an error code would be better here.